internal/statemachine/helper.go

```diff
@@ -864,8 +822,68 @@ func associateLoopDevice(path string, sectorSize quantity.Size) (string, *exec.C
 	return loopUsed, losetupDetachCmd, nil
 }
 
+// DpkgDivert dpkg-diverts the given file in the given baseDir
+// Returns two commands: one for diverting the target file, and one for undiverting it.
+func DpkgDivert(chroot string, target string) (*exec.Cmd, *exec.Cmd) {
+	dpkgDivert := "dpkg-divert"
+	targetDiverted := target + ".dpkg-divert"
+
+	commonArgs := []string{
+		"--local",
+		"--divert",
+		targetDiverted,
+		"--rename",
+		target,
+	}
+
+	divert := append([]string{chroot, dpkgDivert}, commonArgs...)
+	undivert := append([]string{chroot, dpkgDivert, "--remove"}, commonArgs...)
+
+	return execCommand("chroot", divert...), execCommand("chroot", undivert...)
+}
+
 // divertOSProber divert GRUB's os-prober as we don't want to scan for other OSes on
 // the build system
 func divertOSProber(mountDir string) (*exec.Cmd, *exec.Cmd) {
 	return dpkgDivert(mountDir, "/etc/grub.d/30_os-prober")
 }
+
+// DivertExecWithFake replaces a target file in chroot with a provided fake content
+// using dpkg-divert, and returns two commands lists: one for diverting, one for undiverting.
+func DivertExecWithFake(chroot string, file string, fakeContent string) ([]*exec.Cmd, []*exec.Cmd) {
+	divertCmd, undivertCmd := dpkgDivert(chroot, file)
+
+	divertCmds := []*exec.Cmd{
+		divertCmd,
+		execCommand("sh", "-c", "printf '%s' '"+fakeContent+"' > "+filepath.Join(chroot, file)),
+		execCommand("chmod", "+x", filepath.Join(chroot, file)),
+	}
+	undivertCmds := []*exec.Cmd{
+		execCommand("rm", filepath.Join(chroot, file)),
+		undivertCmd,
+	}
+
+	return divertCmds, undivertCmds
+}
+
+// DivertStartStopDaemon diverts [/usr]/sbin/start-stop-daemon with one doing nothing.
+func DivertStartStopDaemon(chroot string) ([]*exec.Cmd, []*exec.Cmd) {
```

Comment:
It depends on where these functions will be in the end, but if they stay in the `statemachine` package they can probably be unexported.